implemented brier score

nicholasloveday opened the pull request:
Still to do

- [x] tests for exceptions
- [x] Jupyter notebook
- [x] Update docs

DerynGriffiths commented:
Hi @nicholasloveday. The Brier Score is suitably brief for such a well known and used measure. A few suggestions. None necessary.
The cell numbering is a bit weird (14, 19, 20 on the preview). I don't know if that matters.
In the first section, it would be worth saying that the Brier Score is proper. Also, smaller is better, with a perfect forecast scoring 0 and the worst score being 1. 
Your example is OK. An alternative is to have forecast skill decreasing with longer lead days (as I did for the flip-flop) and show Brier Score as a function of lead-day. Your choice. Following comments assume you stick with the current example.
Markdown cell describing fcst1 and fcst2 wording could be a bit more elegant. E.g. "By design, fcst1 is better than fcst2. We measure the difference in skill by calculating the Brier Score."
When you print the result I think it is neater to include rounding. In this case, to 2 decimal places should be sufficient and give a nicer printout.
Markdown below that could be more elegantly worded as "As expected, fcst1 has the lower Brier Score quantifying the degree to which it is better than fcst2."

nicholasloveday commented:
Thanks @DerynGriffiths. I've addressed your feedback, but didn't expand the example to work across 7 lead days. I think the Brier score is simple enough that we don't need to expand the example.

DerynGriffiths commented:
Looks good to me.
